cli/pcluster/createami.py

```diff
@@ -111,6 +113,58 @@ def _get_cookbook_dir(region, template_url, args, tmpdir):
         sys.exit(1)
 
 
+def _is_valid_post_install_script(post_install_script_url):
+    return urlparse(post_install_script_url).scheme in ["s3", "https", "file"]
+
+
+def _get_current_timestamp():
+    return datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+
+
+def _get_post_install_script_dir(post_install_script_url, tmp_dir):
+    LOGGER.info("Post install script url: %s", post_install_script_url)
+    try:
+        tmp_post_install_script_folder = os.path.join(tmp_dir, "script")
+        os.mkdir(tmp_post_install_script_folder)
```

Comment:
We're using `tmp_post_install_script_folder` only if `post_install_script_url` is defined so we can move this creation into the `if post_install_script_url` block

Reply:
We use` tmp_post_install_script_folder` to create a temp folder script. Even if `post_install_script_url` is None, we still need to create this temp folder and upload it to packer instance because of consistency. So I think we should leave this out of  `if post_install_script_url` block